### DISCUSSION TITLE
is there anyway to reference the image from assets directory in nuxt3?

### DISCUSSION THREAD

**Discussion by eazypau** [nuxt/nuxt#17399] [Questions] [answered]
When I insert the image path from the assets directory directly <img src="~/assets/some-img-name.png" />, the image will render. However when I insert the image path using a variable. It doesn't render.
For example,
**Child component**
<img :src="imgProp" />
prop: imgProp

**Parent component**
<Child :imgProp="imgUrl" />
const imgUrl = "~/assets/some-img.png"

Am I doing it wrong? or is there another way to do it?

**Comment by benoitdemaegdt** [comment]
I have the same problem.

Did you find a way to solve this ? :)

**Comment by Diizzayy** [comment]
@eazypau @benoitdemaegdt 

Currently, the asset must be referenced directly in the `src` attribute.

if a **variable** with the image asset path is passed to src attribute, It **will not be properly transformed**, and would be rendered in your html as `<img src="~/assets/images/logo.jpg">` as opposed to the correct path to your image.

Example:
```diff
<template>
-  <img :src="logoPath" />
+  <img :src="logoAssetPath" />
+  <img src="~/assets/images/logo.jpg" />
</template>

<script setup>
- const logoPath = `~/assets/images/logo.jpg`; // will not be transformed to correct path
+ const logoAssetPath = '/_nuxt/assets/images/logo.jpg'; // direct reference to the asset, will only work in dev
</script>
```

**Comment by eazypau** [comment, marked as answer]
@Diizzayy 

thanks for the reply. However, this way of writing only works on the development side. If you try to run build and run preview, you will notice that the image is ignore during the build step because the configuration system doesn't know that the particular image is being in use (correct me if I'm wrong coz I also not sure how to explain it in a better/correct manner ).

@benoitdemaegdt 

so far the only way I can think of is either:
- you save your images in public and access it thru the public folder (because it will surely be included in the build step)
you can refer to nuxt documentation on the file reference format for images in public directory (https://nuxtjs.org/docs/directory-structure/static/)
Remark: last time I avoid using it due to they have some minor issue regarding on the src link that is being assigned after the build step but I think the team have resolved the issue

or

- you can statically import the images into the script tag (you can refer to vite document: https://vitejs.dev/guide/assets.html#importing-asset-as-url)
Example:
```
<template>
    <div>
        <img :src="imgLogo" alt="">
    </div>
</template>
<script setup>
    import peopleImg from "~/assets/people-v2.jpg"
    const imgLogo = ref(peopleImg)
</script>
```

Hope it helps :)

**Comment by Sandlens** [comment]
I am having the same issue (rc3). Has this not been fixed yet?
